app/ts/store/transfer.ts

```diff
@@ -0,0 +1,70 @@
+import { signal } from '@preact/signals'
+import { ethers } from 'ethers'
+import { assertsExternalProvider, assertsTransactionHash } from '../library/utilities.js'
+import { TransactionReceipt, TransactionResponse } from '../types.js'
+
+type TransactionRequest = {
+	to: `0x${string}` | string
+	amount: string
+}
+
+export type TransferTransaction =
+	| {
+			status: 'idle'
+			fetchTransactionByHash: (hash: string) => Promise<void>
+			createNewTransfer: () => void
+	  }
+	| {
+			status: 'new'
+			transactionRequest: TransactionRequest
+			sendTransaction: () => Promise<void>
+	  }
+	| {
+			status: 'signed'
+			transactionResponse: TransactionResponse
+			fetchTransactionReceipt: () => Promise<void>
+	  }
+	| {
+			status: 'confirmed'
+			transactionResponse: TransactionResponse
+			transactionReceipt: TransactionReceipt
+	  }
```

Comment:
Similar question to elsewhere.  Why not `useAsyncState`?  Is this different in some significant way?

Reply:
this was the exploration of the use of signal with async stuff "globally" which on the discord discussion should be avoided. adding this to the issue list on #54